verifying/CMakeLists.txt

```
@@ -4,10 +4,40 @@ include_directories(specs)
 set (PASS YieldPass)
 set (PASS_PATH ${CMAKE_BINARY_DIR}/codegen/lib${PASS}.so)
 
+set(CLANG_TOOL ClangPassTool)
+set(CLANG_TOOL_EXECUTABLE ${CMAKE_BINARY_DIR}/clangpass/${CLANG_TOOL})
+
 find_package(Boost REQUIRED COMPONENTS context)
 
+# prefix the actual source file name with '__tmp_' which will be generate by clang pass tool
+set(CLANG_TOOL_TMP_PREFIX "__tmp_")
+
 function(verify_target target)
-    add_executable(${target} ${source_name})
+    if (APPLY_CLANG_TOOL)
+        add_executable(${target} __tmp_${source_name})
+        # apply clangpass to the ${source_name} file
+        add_custom_command(
+            OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/__tmp_${source_name}
+            COMMAND ${CLANG_TOOL_EXECUTABLE}
+                    -p=${CMAKE_BINARY_DIR}/compile_commands.json # passing compilation database, make sure CMAKE_EXPORT_COMPILE_COMMANDS flag is set
+                    --temp-prefix   ${CLANG_TOOL_TMP_PREFIX}
+                    --replace-names ::std::mutex,::std::shared_mutex
+                    --insert-names  ltest::mutex,ltest::shared_mutex
+                    ${CMAKE_CURRENT_SOURCE_DIR}/${source_name}
+            DEPENDS ${CLANG_TOOL}
+            COMMENT "Running Clang Pass Tool on ${source_name}"
+        )
+
+        # delete the temp file generated by previous command
+        add_custom_command(
+            TARGET ${target}
+            POST_BUILD
+            COMMAND ${CMAKE_COMMAND} -E echo "Removing temporary file '${CMAKE_CURRENT_SOURCE_DIR}/${CLANG_TOOL_TMP_PREFIX}${source_name}' generated after building ${target}"
+            COMMAND ${CMAKE_COMMAND} -E remove ${CMAKE_CURRENT_SOURCE_DIR}/__tmp_${source_name}
```

```suggestion
            COMMAND ${CMAKE_COMMAND} -E remove ${CMAKE_CURRENT_SOURCE_DIR}/${CLANG_TOOL_TMP_PREFIX}${source_name}
```

что-то много где одно и тоже встречается, можно переменную тут завести, чтобы покороче писать